### PR TITLE
fix: correct outward supply details using gst treatment

### LINKED ISSUE #1630: Showing incorrect value in GSTR3B outward supplies
### Information about bug

Showing incorrect value in GSTR3B. Also there is no Journal Entry created in this month.
_GSTR1 & Sales Register are showing correct values._

image

image

image

### Version

image

### Relevant log output / Stack trace / Full Error Message.

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ljain112**
Closes: #1630

### CODE REVIEW

File: india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py [hunk 1]
```diff
@@ -296,8 +293,92 @@ def get_inward_nil_exempt(self, state):
...
+            invoice_items = {}
+
+            # Initialize invoice items with default values
+            for item in details["items"]:
+                invoice_items.setdefault(
+                    item.item_code,
+                    {
+                        "taxable_value": 0,
+                        "gst_treatment": item.gst_treatment,
+                        **item_defaults.copy(),
```
Comment (vorasmit) [inline india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py]: no need to copy

File: india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py [hunk 2]
```diff
@@ -346,236 +422,112 @@ def get_outward_items(self, doctype):
...
+                section = self.report_dict["sup_details"][section_head]
+
+                # updating taxable value and tax value
+                section["txval"] += flt(taxable_value, 2)
+                for key in section:
+                    if key in keys_to_update:
+                        section[key] += flt(item.get(key, 0), 2)
+
+                # section 3.2 details
+                if gst_treatment_section:
```
Comment (vorasmit) [inline india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py]: if not taxable

File: india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py [hunk 3]
```diff
@@ -346,236 +422,112 @@ def get_outward_items(self, doctype):
...
-                        self.items_based_on_tax_rate[invoice][0].append(item)
+        return tax_details
 
     def set_outward_taxable_supplies(self):
         inter_state_supply_details = {}
+        gst_treatment_map = {
+            "Nil-Rated": "osup_nil_exmp",
+            "Exempted": "osup_nil_exmp",
+            "Zero-Rated": "osup_zero",
+            "Non-GST": "osup_nongst",
```
Comment (vorasmit) [inline india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py:462]: taxable

File: india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py [hunk 4]
```diff
@@ -346,236 +422,112 @@ def get_outward_items(self, doctype):
...
+
+        for inv, invoice_details in self.invoice_map.items():
+            items = self.invoice_item_wise_tax_details.get(inv, {})
             gst_category = invoice_details.get("gst_category")
             place_of_supply = (
                 invoice_details.get("place_of_supply") or "00-Other Territory"
             )
-            is_overseas_invoice = is_overseas_transaction(
-                "Sales Invoice", gst_category, place_of_supply
+            is_inter_state = gst_category == "SEZ" or (
```
Comment (vorasmit) [inline india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py:465]: use same functionality

File: india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py [hunk 5]
```diff
@@ -346,236 +422,112 @@ def get_outward_items(self, doctype):
...
+                gst_treatment_section = gst_treatment_map.get(item.get("gst_treatment"))
+                taxable_value = item.get("taxable_value")
+                section_head = gst_treatment_section or "osup_det"  # default taxable
+                section = self.report_dict["sup_details"][section_head]
+
+                # updating taxable value and tax value
+                section["txval"] += flt(taxable_value, 2)
+                for key in section:
+                    if key in keys_to_update:
+                        section[key] += flt(item.get(key, 0), 2)
```
Comment (vorasmit) [inline india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py]: .get

File: india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py [hunk 6]
```diff
@@ -346,236 +422,112 @@ def get_outward_items(self, doctype):
...
-                                taxable_value * rate / 100, 2
-                            )
-                            self.report_dict["sup_details"]["isup_rev"]["txval"] += flt(
-                                taxable_value, 2
-                            )
+        section = self.report_dict["sup_details"]["isup_rev"]
+        for inv, invoice_details in self.invoice_map.items():
+            items = self.invoice_item_wise_tax_details.get(inv, {})
+            for item in items.values():
+                section["txval"] += flt(item.get("taxable_value"), 2)
```
Comment (vorasmit) [inline india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py]: += flt

**Comment by ljain112** [comment]
@vorasmit just confirming 0.0 will valid value ??

File: india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py [hunk 7]
```diff
@@ -326,18 +415,13 @@ def get_outward_tax_invoices(self, doctype, reverse_charge=None):
...
         if not self.invoice_map:
-            return
+            return {}
 
         item_details = frappe.db.sql(
             f"""
             SELECT
-                item_code, parent, taxable_value, item_tax_rate,
-                gst_treatment
+                item_code, parent, taxable_value, gst_treatment
```
Comment (vorasmit) [inline india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py]: item_name

**Pull request merged**